extensions/ql-vscode/src/view/webview.tsx

```diff
@@ -25,11 +26,11 @@ const render = () => {
   // eslint-disable-next-line @typescript-eslint/no-var-requires
   const view: WebviewDefinition = require(`./${viewName}/index.tsx`).default;
 
-  ReactDOM_render(
-    view.component,
-    document.getElementById("root"),
-    // Post a message to the extension when fully loaded.
-    () => vscode.postMessage({ t: "viewLoaded", viewName }),
+  createRoot(element).render(
+    // Post a message to the extension when fully loaded. See https://github.com/reactwg/react-18/discussions/5 ("What about the render callback?")
```

Comment:
I'm not sure we need the comment to stick around. We could have it in the commit message for people to find if they want to know why we made the change.

Reply:
I left the comment here because this is quite a strange use of the `ref` property. `ref` is usually used to get a reference to the underlying HTML element, and we're using it in a completely different way. I can definitely remove it and we can rely on the Git history.

Reply:
That's fair, I didn't clock from the link that `ref` is required now. Should we extract this div into an `App` component to more closely match the docs example?